### PR TITLE
Added Secure NATS to DataMiner hardening guide. 

### PR DESCRIPTION
**Pull request by FlavioJA**

### CODE REVIEW

File: user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+       key_file:  "absolute\\path\\to\\private-key_file\\private-key.pem"
+       ca_file:   "absolute\\path\\to\\rootCA_file\\rootCA.pem"
+     }
+   }
+   ```
+
+   > [!NOTE]
+   > The `ca_file` property is only required when self-signed certificates are used.
+
+1. Stop your NATS service.
```
[the remark this answers is not in the document]
Comment (FlavioJA) [reply user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md]: Added details to the services steps.

File: user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+1. Request or generate a TLS certificate (in PEM format).
+
+   To generate self-signed certificates, we recommend that you **use our [scripts for generating TLS certificates](https://github.com/SkylineCommunications/generate-tls-certificates)**, available on GitHub. There is a version of the script for Linux and for Windows machines. The script requires two tools: *openssl* and the *Java keytool*. Both of these can run on Linux and Windows.
+
+1. Update the **nats-server.config** file to incorporate the TLS section within the cluster segment.
+
+   This adjustment is necessary to accommodate the specified location of the PEM files. For example:
+
+   ```
```
[the remark this answers is not in the document]
Comment (FlavioJA) [reply user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md:24]: This it's a type of code on it's own as it's a specific language that is adopted in the NATS configuration file.

File: user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+By default, NATS does **not** employ TLS encryption, leaving communication susceptible to eavesdropping. Consequently, we **strongly recommend enabling TLS encryption** for enhanced security within your NATS cluster.
+
+> [!NOTE]
+> This applies solely to instances involving a DMS cluster or external DxMs. If you have a single-Agent setup, all communication is confined to the local host, as there is only one NATS node.
+
+To enable authentication, please apply the following procedure to all the NATS nodes in the cluster:
+
+1. Request or generate a TLS certificate (in PEM format).
+
+   To generate self-signed certificates, we recommend that you **use our [scripts for generating TLS certificates](https://github.com/SkylineCommunications/generate-tls-certificates)**, available on GitHub. There is a version of the script for Linux and for Windows machines. The script requires two tools: *openssl* and the *Java keytool*. Both of these can run on Linux and Windows.
```
[the remark this answers is not in the document]
Comment (FlavioJA) [reply user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md]: The Linux indication is present because the script does not need to run in the machine where DataMiner is installed, meaning that the user can decide to run it on Linux and then copy over the generated certificates to the Windows machine.

File: user-guide/Advanced_Functionality/Security/DataMiner_hardening_guide.md [hunk 4]
```diff
@@ -53,6 +53,13 @@ For the inter-DMA communication, like for the communication with DataMiner Cube,
 
 To enable gRPC for the communication between DataMiner Agents in a cluster, add [redirects in DMS.xml](xref:DMS_xml#redirects-subtag) or, from **10.3.6/10.3.0 [CU3] onwards**, disable .NET Remoting completely in [MaintenanceSettings.xml](xref:Configuration_of_DataMiner_processes#disabling-net-remoting).
 
+#### NATS
+
+By default, NATS does not employ TLS encryption, leaving communication susceptible to eavesdropping. Consequently, we strongly recommend [enabling TLS encryption for enhanced security within your NATS cluster](xref:Security_NATS).
+
+> [!IMPORTANT]
+> TLS encryption over NATS is exclusively accessible starting from DataMiner 10.4.3 CU0 onwards when employing CA-Signed Certificates (Certificate Authority-Signed Certificates). It's crucial to be aware that the utilization of Self-Signed Certificates may result in DataMiner startup failures.
```
[the remark this answers is not in the document]
Comment (FlavioJA) [reply user-guide/Advanced_Functionality/Security/DataMiner_hardening_guide.md:61]: The release note that adds this feature is [RN38302](https://intranet.skyline.be/DataMiner/Lists/Release%20Notes/DispForm2.aspx?ID=38302).

To clarify, the 10.4.3 version, which is now sealed is working with trusted CA-Signed Certificates, while the 10.4.4 version will work with self-signed certificates. Please note that the 10.4.3 would in theory already work with self-signed certificates, which was unfortunately not possible due to development issues.
Therefore the procedure for the self-signed was already created in the Secure NATS section, which can be adjusted in order to indicate that or it can directly indicated in this page.

[the remark this answers is not in the document]
Comment (MariekeGO) [reply user-guide/Advanced_Functionality/Security/DataMiner_hardening_guide.md]: @FlavioJA, when I check the description of RN 38302, this seems to apply to the BrokerGateway soft-launch feature only: "*The MessageBroker now supports TLS (Transport Layer Security) configuration, enhancing the security of communication between NATS clients and the NATS server. This feature is available when using BrokerGateway*". 
Is this indeed only available if the *BrokerGateway* soft-launch option is enabled? Or is this actually a general change, despite what is suggested in the RN?
If it is a soft-launch feature, I'm not sure we should add this PR in its current form yet, because we only allow the use of soft-launch features on staging platforms, but users reading the new page will assume they need to apply this on their production platforms as well.

[the remark this answers is not in the document]
Comment (FlavioJA) [reply user-guide/Advanced_Functionality/Security/DataMiner_hardening_guide.md]: The BrokerGateway itself is a soft-launch option. However, it is not really needed to achieve TLS over NATS if the nats-server.config is configured manually, as we describe in the Secure NATS section.

**Comment by MariekeGO** [comment]
@SeppeDejonckheere, as this is related to security, could you take a look at this PR and confirm if this is OK to merge?

File: user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+          ],
+          "tls_required": true,
+          "tls_verify": true
+        }
+      ```
+
+## Enabling DataMiner-to-NATS node TLS communication
+
+> [!IMPORTANT]
+> TLS encryption for communication between DataMiner and NATS nodes is supported from DataMiner 10.4.3 [CU0]/10.5.0 onwards<!-- RN 38302 -->, with CA-signed certificates (Certificate Authority-Signed Certificates). Avoid using self-signed certificates for this, as this may result in DataMiner startup failures.
```
Comment (SeppeDejonckheere) [inline user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md:68]: I would rephrase 'Avoid using self-signed certificates for this, as this may result in DataMiner startup failures.' to something along the line of 'For DataMiner 10.4.3, only CA-signed certificates are supported. Using a self-signed certificate will cause DataMiner to fail to start up'.

This specific issue with self-signed certs should be fixed from 10.4.4 onwards.

**Review by SeppeDejonckheere — APPROVED** [review]

**Review by MariekeGO — APPROVED** [review]

File: user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md (hunk 5 above)
Comment (MariekeGO) [reply user-guide/Advanced_Functionality/Security/Advanced_security_configuration/Security_NATS.md]: I've rephrased the note so we don't promise an exact version with the fix, as I haven't seen a 10.4.4 RN for this yet. As soon as there is an RN for this, we can update this.
